pkg/csi_driver/multishare_ops_manager.go

```diff
@@ -651,16 +647,69 @@ func containsOpWithInstanceTargetPrefix(instance *file.MultishareInstance, ops [
 	return nil, nil
 }
 
-func (m *MultishareOpsManager) listInstanceForStorageClassPrefix(ctx context.Context, prefix string) ([]*file.MultishareInstance, error) {
+// listMatchedInstances will list all instances in current project,
+// but only matched instances will be returned.
+func (m *MultishareOpsManager) listMatchedInstances(ctx context.Context, req *csi.CreateVolumeRequest, target *file.MultishareInstance) ([]*file.MultishareInstance, error) {
 	instances, err := m.cloud.File.ListMultishareInstances(ctx, &file.ListFilter{Project: m.cloud.Project, Location: "-"})
 	if err != nil {
 		return nil, err
 	}
 	var finalInstances []*file.MultishareInstance
 	for _, i := range instances {
-		if val, ok := i.Labels[util.ParamMultishareInstanceScLabelKey]; ok && val == prefix {
+		matched, err := isMatchedInstance(i, target, req)
+		if err != nil {
+			return nil, err
+		}
+		klog.Infof("Found source instance %+v, comparing with target instance %+v and StorageClass parameters %v, matched = %t", *i, *target, req.GetParameters(), matched)
```

Comment:
`Found source instance {Project:tyuchn-test Name:fs-a4e913ef-d4d1-496c-8d31-150b32b01a9e Location:us-central1 Tier:ENTERPRISE Network:{Name:projects/mattcary-saikatroyc-test/global/networks/shared-net ConnectMode:PRIVATE_SERVICE_ACCESS ReservedIpRange:10.155.16.0/26 Ip:10.155.16.2} CapacityBytes:1099511627776 MaxCapacityBytes:10995116277760 CapacityStepSizeGb:256 Labels:map[storage_gke_io_created-by:filestore_csi_storage_gke_io storage_gke_io_storage-class-id:csi-filestore-multishare-1] State:READY KmsKeyName: Description:}, comparing with target instance {Project:tyuchn-test Name:fs-0cae12d9-04a3-4585-a9be-6583bc212d6b Location:us-central1 Tier:enterprise Network:{Name:default ConnectMode:DIRECT_PEERING ReservedIpRange: Ip:} CapacityBytes:1099511627776 MaxCapacityBytes:0 CapacityStepSizeGb:0 Labels:map[storage_gke_io_created-by:filestore_csi_storage_gke_io storage_gke_io_storage-class-id:csi-filestore-multishare] State: KmsKeyName: Description:} and StorageClass parameters map[csi.storage.k8s.io/pv/name:pvc-8080db2f-08fe-4511-92ec-92cf176d326c csi.storage.k8s.io/pvc/name:test-pvc-multishare csi.storage.k8s.io/pvc/namespace:default instance-storageclass-label:csi-filestore-multishare multishare:true tier:enterprise], matched = false`